components/ConfirmationCodeInput.js

```diff
@@ -209,32 +211,55 @@ export default class ConfirmationCodeInput extends Component {
     }
   }
   
-  _onInputCode(character, index) {
+  _onInputCode(character, i) {
     const { codeLength, onFulfill, compareWithCode, ignoreCase, onCodeChange } = this.props;
     let newCodeArr = _.clone(this.state.codeArr);
+    let index = i;
+    let isPasteAction = false;
     newCodeArr[index] = character;
-    
-    if (index == codeLength - 1) {
-      const code = newCodeArr.join('');
-      
-      if (compareWithCode) {
-        const isMatching = this._isMatchingCode(code, compareWithCode, ignoreCase);
-        onFulfill(isMatching, code);
-        !isMatching && this.clear();
-      } else {
-        onFulfill(code);
+    if (character.length > codeLength) {
+      return;
+    }
+    if (index === 0 && character.length > 1) {
+      newCodeArr = character.split('');
+      if (character.length < codeLength) {
+        this._setFocus(codeLength -1);
+      } else if (character.length === codeLength) {
+        this._blur(index)
       }
-      this._blur(this.state.currentIndex);
-    } else {
-      this._setFocus(this.state.currentIndex + 1);
+      index = character.length -1;
+      isPasteAction = true;
```

Comment:
flag to prevent focus, active and set a timeout to allow the ui to show the fields completed.